src/app/donation-start/donation-start-form/donation-start-form.component.ts

```diff
@@ -507,6 +507,32 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
     const stepperHeaders = stepper.getElementsByClassName('mat-step-header');
     for (const stepperHeader of stepperHeaders) {
       stepperHeader.addEventListener('click', (clickEvent: any) => {
+        // Disallow any step jumps by header click without a valid captcha, since the donor could have
+        // dismissed the puzzle (including by accident) and they'll face errors later, which are more annoying,
+        // without a working code. If they click step 1 while on step 1 this might fire the captcha slightly
+        // earlier than normal, but that should be mostly harmless – at worst if they take that unusual
+        // step they might have to solve 2 puzzles.
+
+        // For now, I've gone with only checking this on header click because the other way of changing step
+        // is, as far as we know, working. And we want to minimise the change during CC23. A better solution
+        // later might be to remove all the captcha executes that happen specifically on button clicks (`next()`)
+        // and to execute it in `stepChanged()` – except when just booted back to step 0 – instead.
+        if (!this.idCaptchaCode) {
+          clickEvent.preventDefault();
+
+          this.stepChangeBlockedByCaptcha = true;
+          try {
+            this.idCaptcha.reset();
+          } catch (e) {
+            this.matomoTracker.trackEvent('identity_error', 'step_change_captcha_reset_failed', e.message);
+            this.reset(); // Includes page refresh atm – annoying but less so than filling out even more broken fields.
+            return;
+          }
+
+          this.idCaptcha.execute();
```

Comment:
Setting `stepChangeBlockedByCaptcha` will make the stepper jump to the next step on captcha return. But we don't want the next step here, do we? We want whichever step they clicked on. Do we need to store that step id somewhere to use it in the captchaReturn?

Reply:
Happy to look at the docker stuff tomorrow with you if you like. Not sure what the solution is.